[Bug] struct drop order

## Description
Codegen generates struct `_drop` functions in declaration order, causing implicit declaration errors when a struct references another struct defined later in the file.

## Steps to Reproduce
1. Create file with struct used as field type in another struct
2. Run `urusc file.urus`
3. Compile fails with C compiler error
4. If we reorder the struct in another order, it produces this warning #53 

## Expected Behavior
Should compile successfully - forward declarations should handle this.

## Actual Behavior
C compiler error: implicit declaration of function 'Point_drop'

## Code (if relevant)
```rust
struct Circle {
    center: Point;
    radius: float;
}

struct Point {
    x: float;
    y: float;
}

fn main(): void {
    let p1: Point = Point { x: 0.0, y: 0.0 };
    let c: Circle = Circle { center: p1, radius: 5.0 };
    print(c.radius);
}

```

## Error Message
```
_urus_tmp.c: In function 'Circle_drop':
_urus_tmp.c:531:9: error: implicit declaration of function 'Point_drop' [-Wimplicit-function-declaration]
  531 |         Point_drop(&(*obj)->center);
      |         ^~~~~~~~~~
_urus_tmp.c: At top level:
_urus_tmp.c:537:13: warning: conflicting types for 'Point_drop'; have 'void(Point **)'
  537 | static void Point_drop(Point **obj) {
      |             ^~~~~~~~~~
_urus_tmp.c:537:13: error: static declaration of 'Point_drop' follows non-static declaration
  537 | static void Point_drop(Point **obj) {
      |             ^~~~~~~~~~
```

## Root Cause
In `codegen.c`, the `_drop` functions are generated in the same pass as struct definitions. When `Circle_drop` references `Point_drop`, `Point` hasn't been defined yet.

## Environment
- OS: Linux (Arch)
- GCC version: 15.2.1
- URUS version: 0.2.2 (F)-Fixed

--- a/compiler/src/codegen.c
+++ b/compiler/src/codegen.c
@@ -904,7 +904,24 @@ void codegen_generate(CodeBuf *buf, AstNode *program) {
     }
     emit(buf, "\n");
 
-    // Pass 3B: struct drop function
+    // Pass 3B: struct drop forward declarations
+    for (int i = 0; i < program->as.program.decl_count; i++) {
+        AstNode *d = program->as.program.decls[i];
+        if (d->kind == NODE_STRUCT_DECL) {
+            emit(buf, "static void %s_drop(%s **obj);\n", d->as.struct_decl.name, d->as.struct_decl.name);
+        }
+    }
+
+    // Pass 3C: enum drop forward declarations
+    for (int i = 0; i < program->as.program.decl_count; i++) {
+        AstNode *d = program->as.program.decls[i];
+        if (d->kind == NODE_ENUM_DECL) {
+            emit(buf, "static void %s_drop(%s **obj);\n", d->as.enum_decl.name, d->as.enum_decl.name);
+        }
+    }
+    emit(buf, "\n");
+
+    // Pass 3D: struct drop function
     for (int i = 0; i < program->as.program.decl_count; i++) {
         AstNode *d = program->as.program.decls[i];
         if (d->kind == NODE_STRUCT_DECL) {
@@ -928,7 +945,7 @@ void codegen_generate(CodeBuf *buf, AstNode *program) {
         }
     }
 
-    // Pass 3C: enum drop function
+    // Pass 3E: enum drop function
     for (int i = 0; i < program->as.program.decl_count; i++) {
         AstNode *d = program->as.program.decls[i];
         if (d->kind == NODE_ENUM_DECL) {

